src/components/Dropdown/Dropdown.jsx

```diff
@@ -1,36 +1,47 @@
+import { forwardRef } from "react";
+
 import arrowDown from "../../assets/icons/arrow-down.svg";
 import arrowTop from "../../assets/icons/arrow-top.svg";
 
 import styles from "./Dropdown.module.css";
 
-const Dropdown = ({ selectedText, isOpen, setIsOpen, items, onSelect }) => {
-  return (
-    <div className={styles.dropdownContainer}>
-      <div
-        role="button"
-        className={styles.dropdown}
-        onClick={() => setIsOpen((prev) => !prev)}
-      >
-        <span>{selectedText}</span>
-        <div className={styles.iconDiv}>
-          <img src={isOpen ? arrowTop : arrowDown} alt="arrow-icons" />
+const Dropdown = forwardRef(
+  ({ selectedText, isOpen, setIsOpen, items, onSelect }, ref) => {
+    const handleDropdownClick = (e) => {
+      if (e.target !== e.currentTarget) {
```

Comment:
단순히 클릭한 요소와 이벤트 핸들러 전달된 요소가 같지 않다고만 표현하면,
드롭다운 버튼의 자식요소를 클릭했을 때에도 이벤트가 발생하지 않아요. 🤔

아래처럼 자식 요소에 모두 포함되는지 여부를 판단하는 게 좋을 것 같아요. 😄
```suggestion
      if (e.currentTarget.contains(e.target)) {
```